packages/editor/src/client/blocks/mapLeafletBlock.tsx

```diff
@@ -0,0 +1,168 @@
+import React, {useState} from 'react'
+import {Map, TileLayer, Marker, Popup} from 'react-leaflet'
+import 'leaflet/dist/leaflet.css'
+import '../customCSS/leaflet.css'
+
+import {Icon} from 'leaflet'
+
+// stupid hack so that leaflet's images work after going through webpack
+//@ts-ignore
+import marker from 'leaflet/dist/images/marker-icon.png'
+//@ts-ignore
+import marker2x from 'leaflet/dist/images/marker-icon-2x.png'
+//@ts-ignore
+import markerShadow from 'leaflet/dist/images/marker-shadow.png'
+
+//@ts-ignore
+delete Icon.Default.prototype._getIconUrl
+
+Icon.Default.mergeOptions({
+  iconRetinaUrl: marker2x,
+  iconUrl: marker,
+  shadowUrl: markerShadow
+})
+
+import {
+  BlockProps,
+  FieldProps,
+  Box,
+  Card,
+  PlaceholderInput,
+  ZIndex,
+  IconButton,
+  Spacing,
+  TypographicTextArea,
+  Drawer,
+  Image
+} from '@karma.run/ui'
+
+import {
+  MaterialIconImageOutlined,
+  MaterialIconEditOutlined,
+  MaterialIconClose
+} from '@karma.run/icons'
+
+import {ImageSelectPanel} from '../panel/imageSelectPanel'
+import {ImagedEditPanel} from '../panel/imageEditPanel'
+
+import {MapLeafletBlockValue, MapLeafletItem} from './types'
+
+// import {isFunctionalUpdate} from '@karma.run/react'
+
+export interface MapLeafletBlockProps extends BlockProps<MapLeafletBlockValue> {}
+
+export function MapLeafletBlock({value, onChange, disabled}: BlockProps<MapLeafletBlockValue>) {
+  const {zoom, centerLat, centerLng, caption} = value
+
+  return (
+    <>
+      <Map
+        center={[centerLat, centerLng]}
+        zoom={zoom}
+        style={{width: '100%', height: '45vh'}}
+        onViewportChange={e => {
```

Comment:
in general i would not work with single letter variables. There's no gain in shortening the variable and other devs have a harder time understanding it.

Reply:
good point, I replaced it with *viewport*